topic-configuration/README.adoc

```diff
@@ -0,0 +1,689 @@
+|===
+--
+
+Partitions::
++
+--
+Partitions are distinct lists of messages within a topic. Partitions are the main concurrency mechanism in Kafka and enable parts of a topic to be distributed over multiple brokers in the cluster. A topic can contain one or more partitions, enabling producer and consumer loads to be scaled. After you create a topic, you can increase the number of partitions but you cannot decrease it.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|Yes (increase only)
+
+h|Type
+|Integer
+
+h|Default value
+|`1`
+
+h|Supported values
+|[`1,...`]
+
+h|Kafka property name
+|`num.partitions`
+|===
+--
+
+Replicas::
++
+--
+Replicas are copies of partitions in a topic. Partition replicas are distributed over multiple brokers in the cluster to ensure topic availability if a broker fails. When a follower replica is in sync with a partition leader, the follower replica can become the new partition leader if needed.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Integer
+
+h|Default value
+|`3`
+
+h|Kafka property name
+|`replication.factor`
+|===
+--
+
+Minimum in-sync replicas::
++
+--
+Minimum in-sync replicas is the minimum number of replicas that must acknowledge a write for the write to be considered successful. This property assumes that the producer requests acknowledgments from all replicas (`acks` set to `all`). If this minimum is not met, the producer raises an exception (`NotEnoughReplicas` or `NotEnoughReplicasAfterAppend`).
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Integer
+
+h|Default value
+|`2`
+
+h|Kafka property name
+|`min.insync.replicas`
+|===
+--
+
+Retention time::
++
+--
+Retention time is the amount of time that messages are retained in a topic before they are deleted. This property applies only when the topic cleanup policy is set to `Delete` or `Compact, Delete`.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|Yes
+
+h|Type
+|Long
+
+h|Default value
+|`604800000 ms` (7 days)
+
+h|Supported values
+| `milliseconds`, `seconds`, `days`, `months`, `years`, `Unlimited`
+
+h|Kafka property name
+|`retention.ms`
+|===
+--
+
+Retention size::
++
+--
+Retention size is the maximum total size of all log segments in a partition before old log segments are deleted to free up space. By default, no retention size limit is applied, only a retention time limit. This property applies only when the topic cleanup policy is set to `Delete` or `Compact, Delete`.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|Yes
+
+h|Type
+|Long
+
+h|Default value
+|`Unlimited`
+
+h|Supported values
+| `bytes`, `kilobytes`, `megabytes`, `gigabytes`, `terabytes`, `Unlimited`
+
+h|Kafka property name
+|`retention.bytes`
+|===
+--
+
+=== Messages
+
+These properties control how your messages are handled in the Kafka instance.
+
+Maximum message bytes::
++
+--
+Maximum message bytes is the maximum record batch size.
```

Comment:
1 MB is optimal for maximum throughput in most use cases

Reply:
Ack. I think the idea is that all default values of all properties, especially the properties that users can't change, are set at what we've deemed the optimal setting, given that it's a managed service. So I think I'm okay not to mention it for each property.